lib/ansible/plugins/lookup/dsv.py

```diff
@@ -0,0 +1,128 @@
+       DSV `tenant` using a `client_id` and `client_secret`.
+requirements:
+    - python-dsv-sdk - https://pypi.org/project/python-dsv-sdk/
+options:
+    _terms:
+        description: the path to the secret e.g. /staging/servers/web1
+        required: True
+    tenant:
+        description: the first format parameter in the default `url_template`
+        env:
+            - name: DSV_TENANT
+        ini:
+            - section: dsv_lookup
+              key: tenant
+        required: True
+    tld:
+        default: com
+        description: the top-level domain of the tenant; the second format
+            parameter in the default `url_template`
+        env:
+            - name: DSV_TLD
+        ini:
+            - section: dsv_lookup
+              key: tld
+        required: False
+    client_id:
+        description: the client_id with which to request the Access Grant
+        env:
+            - name: DSV_CLIENT_ID
+        ini:
+            - section: dsv_lookup
+              key: client_id
+        required: True
+    client_secret:
+        description: the client_secret associated with the specific client_id
+        env:
+            - name: DSV_CLIENT_SECRET
+        ini:
+            - section: dsv_lookup
+              key: client_secret
+        required: True
+    url_template:
+        default: https://{}.secretsvaultcloud.{}/v1
+        description: the path to append to the base URL to form a valid REST
+            API request
+        env:
+            - name: DSV_URL_TEMPLATE
+        ini:
+            - section: dsv_lookup
+              key: url_template
+        required: False"""
+
+RETURN = r"""
+_list:
+    description:
+        - One or more JSON responses to `GET /secrets/{path}`
+        - See https://dsv.thycotic.com/api/index.html#operation/getSecret"""
+
+EXAMPLES = r"""
+- hosts: localhost
+  vars:
+      secret: "{{ lookup('dsv', '/test/secret') }}"
+  tasks:
+      - debug: msg="the password is {{ secret["data"]["password"] }}"
+"""
+
+from ansible.errors import AnsibleError, AnsibleOptionsError
+
+try:
+    from thycotic.secrets.vault import (
+        SecretsVault,
+        SecretsVaultError,
+    )
+except ImportError:
+    raise AnsibleError("python-dsv-sdk must be installed to use this plugin")
+
+from ansible.module_utils._text import to_native
+from ansible.utils.display import Display
+from ansible.plugins.lookup import LookupBase
+
+
+display = Display()
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables, **kwargs):
+        self.set_options(var_options=variables, direct=kwargs)
+
+        vault_parameters = {
+            "tenant": self.get_option("tenant"),
+            "client_id": self.get_option("client_id"),
+            "client_secret": self.get_option("client_secret"),
+            "url_template": self.get_option("url_template"),
+        }
+
+        display.vvv("DevOps Secrets Vault parameters: %s" % vault_parameters)
```

Comment:
Do not display vault_parameters to stdout or logs since it contains client_secret. This will lead to CVE.

Reply:
Done.